openevsewifi/__init__.py

```diff
@@ -27,22 +28,65 @@
 
 colors = ['off', 'red', 'green', 'yellow', 'blue', 'violet', 'teal', 'white']
 
+def parse_checksum(s):
+    """
+    If there is a '^' in given string s, this checks that the xor of utf8 bytes
+    before the '^' equal the hex value specified after '^'.  It returns the
+    string before the '^' on success, None on error.
+
+    If there is no '^' in the string, the string is returned.
+    """
+    spl = s.rsplit('^', 1)
+    if len(spl) == 1:
+        return s
+    try:
+        check = int(spl[1], 16)
+    except:
+        return None
+    datsum = 0
+    for c in spl[0].encode('utf-8'):
+        datsum ^= c
+    if datsum != check:
+        return None
+    return spl[0]
+
+def json_parser(s):
+    print("json: ", s)
```

Comment:
Please remove these print statements.  I'm not opposed to a PR that implements proper logging, but this isn't the right way to do it, especially inside a library.

Reply:
sorry. this was spurious debugging that I'd been using before I started working on the tests. b/c the test harness doesn't give stdout, I missed removing them.